src/pages/MoviesPage.vue

```diff
@@ -27,60 +32,92 @@
 import MainTitle from "../components/MainTitle.vue";
 import MovieCard from '../components/MovieCard.vue';
 import BaseInput from '../components/BaseInput.vue';
-import axios from 'axios';
-import {BASE_URL} from '../helpers/index';
+import SelectInput from '../components/SelectInput.vue'
+import ActionBar from '../components/ActionBar.vue';
 
 export default {
-  components: { MainTitle, MovieCard, BaseInput },
+  components: { MainTitle, MovieCard, BaseInput, ActionBar, SelectInput },
 
   data(){
   return {
-    movies: [],
-    errorMsg: '',
     query: '',
+    selected: 'All',
+    
   }
   },
 
   computed:{
-    filteredMovies(){
-    const foundMovies = this.movies.filter(movie => movie.title.toLowerCase()
-    .includes(this.query.toLowerCase()));
-     return foundMovies;
+    movies(){
+      return this.$store.state.movies
+    },
+    filteredByQuery(){
+    return this.movies.filter(movie => movie.title.toLowerCase().includes(this.query.toLowerCase()));
+    },
+
+    filteredByGenre(){
+      const filteredGenre = this.filteredByQuery.filter(movie => movie.genre.name === this.selected);
+
+      return this.selected === 'All' ? this.filteredByQuery : filteredGenre;
     }
   },
 
-  methods:{
-    async fetchData(){
-      try{
-         const response = await axios.get(`${BASE_URL}/movies`);
-         console.log(response.data)
-         this.movies = response.data;
-      } catch{
-          this.errorMsg = 'Something went wrong...'
-          console.log(this.errorMsg)
-      }
-     
-    },
-  
-  },
-  mounted(){
-    this.fetchData()
+  async mounted(){
+    this.$store.dispatch('fetchMovies')
   },
+ 
   
 };
 
 </script>
 
 
 <style lang="scss">
+
+.page-wrapper{
+    @include pageCenter();
+  
+   }
 .movies-container{
   display: flex;
   flex-wrap: wrap;
   justify-content: center;
+  margin-top: 50px;
+  
+  a{
+    text-decoration: none;
+  }
+}
+
+.searchBar{
+  width: 100%;
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+
+
+   &__input{
+    width: 100%;
+   }
+
+  &__select{
+    width: 100%;
+    
+  }
+}
 
-  align-items: center;
-  width: 97%;
-  max-width: 1440px;
-  margin: 0 auto;
+@media(min-width: 785px){
+  .searchBar{
```

Comment:
In CSS classes should be rather written with kebab-case than camelCase. CSS doesn't recognize casing - it would be the same as `.searchbar`